src/bot/data/guild_repository.py

```diff
@@ -1,4 +1,28 @@
+import aiosqlite
+import sqlite3
+import logging
+
 from bot.data.base_repository import BaseRepository
+from bot.errors import PrimaryKeyError
+log = logging.getLogger(__name__)
+
 
 class GuildRepository(BaseRepository):
-    pass
+
+    async def add_guild(self, guild):
+        try:
+            async with aiosqlite.connect(self.database_name) as db:
```

Comment:
It might be good to have the DB provider be specified in the BaseRepository or Database. This allows changing the BaseRepository or Database to use a different backend without having to change all the other files. We could also use something like [pyodbc](https://github.com/mkleehammer/pyodbc/wiki) to provide a common and interchangeable interface for the DB.

Reply:
This is a good idea and we should prob do this, its worth noting that switching to a non SQLite dbms will require alot the schema and the queries to change in all but the simplest cases. so we should hammer this out before we continue to far

Reply:
I don't know if it will have to change that much. Inserts and selects should pretty much all be the same. It just depends on whether we want to do data manipulation on the DB or bot. If we don't need to do any fancy joins or statements we can do all that on the data we get back.

We might want to move this discussion out of this PR to a GH Issue or the discord

Reply:
This def needs to be decided sooner rather then later lol